values/prometheus-operator/prometheus-operator.gotmpl

```diff
@@ -53,6 +54,7 @@ prometheus:
             {{- range $teamId, $_ := $v.teamConfig }}
             - team-{{ $teamId }}
             {{- end }}
+    {{- end }}
```

Comment:
```suggestion
    {{- else }}
    {{ $selType }}NamespaceSelector:
      matchLabels: {}
    {{- end }}
```

Reply:
To my surprise I found that with an empty value it targets only the monitoring namespace, so all-namespace targeting must be explicitly declared I'm afraid. With this commit it would resolve to the same as the chart default. Not ready for merging yet I'm afraid.

Reply:
`matchLabels: {}` is not an empty value though. The logic in the chart:
```yaml
{{- if .Values.prometheus.prometheusSpec.serviceMonitorNamespaceSelector }}
  serviceMonitorNamespaceSelector:
{{ toYaml .Values.prometheus.prometheusSpec.serviceMonitorNamespaceSelector | indent 4 }}
{{ else }}
  serviceMonitorNamespaceSelector: {}
{{- end }}
```
Did you try my suggestion?

Reply:
This will not work, `XXMonitorNamespaceSelector: {}` is the same as the chart default, and will result in Prometheus only picking up monitor objects in the monitoring namespace (the namespace into which Prometheus is deployed). Including your commit or not, it still resolves to `XXMonitorNamespaceSelector: {}`.
Like I said, we must explicitly declare all-namespace targeting, and I will come back to this Monday.

Reply:
Please read carefully:
```yaml
XXMonitorNamespaceSelector: {}
```
is NOT the same as:
```yaml
XXMonitorNamespaceSelector:
  matchLabels: {}
```